vllm/model_executor/layers/linear.py

```diff
@@ -259,14 +263,17 @@ def __init__(self,
         self.gather_output = gather_output
 
         # Divide the weight matrix along the last dimension.
+        tp_rank = get_tensor_model_parallel_rank()
         tp_size = get_tensor_model_parallel_world_size()
         assert self.quant_method is not None
-        self.output_size_per_partition = divide(self.output_size, tp_size)
+        self.output_size_per_partition = output_size // tp_size + (
+            output_size % tp_size > tp_rank)
```

Comment:
Could you separate this into its own variable as the remainder for clarity and/or please add a comment describing what is the intended behavior? The condition makes it a bit unclear

Reply:
Nice catch, this should use the util function I added for this logic, which should make it more readable.